applications/adv_networking_bench/cpp/testing/adv_networking_test_harness.py

```diff
@@ -0,0 +1,193 @@
+          stderr=subprocess.PIPE,
+      )
+    except Exception as e:
+        logger.error(f"Failed to create process: {e}")
+        raise
+
+    # Stream & capture log
+    while p.poll() is None:
+        for line in iter(p.stderr.readline, ""):
+            print(line.strip())
+            stderr_lines.append(line)
+        for line in iter(p.stdout.readline, ""):
+            print(line.strip())
+            stdout_lines.append(line)
+
+    # Capture any remaining output after process ends
+    for line in p.stderr:
+        print(line.strip())
+        stderr_lines.append(line)
+    for line in p.stdout:
+        print(line.strip())
+        stdout_lines.append(line)
+
+    # Close pipes
+    p.stdout.close()
+    p.stderr.close()
+
+    # Return
+    std_out = "".join(stdout_lines)
+    std_err = "".join(stderr_lines)
+    rc = p.poll()
+    if rc != 0:
+        logger.debug(f"Command failed with exit code {rc}")
+        raise CalledProcessError(rc, cmd, std_out, std_err)
+    return CompletedProcess(cmd, rc, std_out, std_err)
+
+
+def validate_ano_benchmark(
+    log: str,
+    error_packets_threshold: int,
+    missed_packets_threshold: int,
+    avg_throughput_threshold: float,
+) -> bool:
+    """
+    Validates the benchmark results.
+
+    Args:
+        log (str): The output of the benchmark run.
+        error_packets_threshold (int): Maximum allowed errored packets.
+        missed_packets_threshold (int): Maximum allowed missed packets.
+        avg_throughput_threshold (float): Minimum required average throughput in Gbps.
+    Returns:
+        bool: True if the benchmark passed, False otherwise.
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug(f"Validating benchmark results")
+
+    success = True
+    # Check for errors in the log
+    if "[error]" in log:
+        logger.error("Errors found in benchmark output")
+        success = False
+
+    # Check for missed packets
+    transmit_packets = int(re.search(r'Port 0.*?Transmit packets:\s+(\d+)', log, re.DOTALL).group(1))
+    received_packets = int(re.search(r'Port 1.*Received packets:\s+(\d+)', log, re.DOTALL).group(1))
+    received_bytes = int(re.search(r'Port 1.*Received bytes:\s+(\d+)', log, re.DOTALL).group(1))
+    missed_packets = int(re.search(r'Port 1.*Missed packets:\s+(\d+)', log, re.DOTALL).group(1))
+    errored_packets = int(re.search(r'Port 1.*Errored packets:\s+(\d+)', log, re.DOTALL).group(1))
+    exec_time = float(re.search(r'TOTAL EXECUTION TIME OF SCHEDULER : (\d+\.\d+) ms', log).group(1))
+
+    logger.debug("Transmit packets:", transmit_packets)
+    logger.debug("Received packets:", received_packets)
+    logger.debug("Received bytes:", received_bytes)
+    logger.debug("Missed packets:", missed_packets)
+    logger.debug("Errored packets:", errored_packets)
+    logger.debug("Exec time:", exec_time)
+
+    # Check for missed packets
+    missed_packets_ratio = (missed_packets / transmit_packets) * 100
```

Comment:
Nitpick: `ratio` implies range `[0,1]`, suggest rename here and below

```suggestion
    missed_packets_percentage = (missed_packets / transmit_packets) * 100
```